src/scores/continuous/flip_flop_impl.py

```diff
@@ -422,135 +420,16 @@ def flip_flop_index_proportion_exceeding(
         `scores.continuous.flip_flop_index`
 
     """
-    if (dims is not None) and (sampling_dim in list(dims)):
+    if (preserve_dims is not None) and (sampling_dim in list(preserve_dims)):
```

Comment:
What about if `reduce_dims` is not None, and a forecast dim that is not listed in `reduce_dims` is the same as `sampling_dim`?

Reply:
I'm not sure what you exactly mean by this but I'd added a `raises` statement in `flip_flop_index_proportion_exceeding` to throw an error if `reduce_dims` contains `sampling_dim`. Other than that the function just passes through the `preserve_dims` and `reduce_dims` arguments to `gather_ dims` so there's not much need to unit test cases other than if the function throws the correct errors, which I've done. Its up to you on whether you want to throw an error or pop the `sampling_dim` value out of the `reduce_dims` list which you can change as to what makes the most sense.

See source code here: b86045f

Reply:
I'm suggesting the opposite of what you just implemented. Instead of 
```python
    if reduce_dims is not None and sampling_dim in list(reduce_dims):
        raise DimensionError(
            f"`sampling_dim`: '{sampling_dim}' must not be in dimensions to reduce "
            f"`reduce_dims`: {list(reduce_dims)}"
        )
```
I think it should be
```python
    if reduce_dims is not None and sampling_dim is not in list(reduce_dims):
        raise DimensionError(
            f"`sampling_dim`: '{sampling_dim}' must be in dimensions to reduce "
            f"`reduce_dims`: {list(reduce_dims)}"
        )
```

We need to add a test for `flip_flop_index_proportion_exceeding` where the `reduce_dims` arg is called and not just the `preserve_dims` arg

Reply:
Ignore that. How you implemented it was correct since `gather_dimensions` is called after `flip_flop_index` has been executed (and it makes sense to do this check before `flip_flop_index` is called.